Chrysalis-Papilio/Soen-490-Papilio
# Papilio-7 User info is stored in the database

# General info

**Issue number**: PAPILIO-7

**Task description**: 

 - Logging user information to the database
 - Added progress bar to show the user some visual feedback as to what is going on
 - Better error handling when using firebase services
 - Setup retrofit library to make network calls

# Manual Testing
To test this functionality you have to:

1. Add this line to your local.properties file located in the root of the project: 
```
BACKEND_API_URL=http://192.168.1.36:1337/api/user/
The ip address should match your computer's IP address in the form shown above (**not** localhost or the loopback address)
```
2. Add the following line to the network_security_config.xml file:

```
<domain includeSubdomains="true">192.168.1.36</domain>
The IP address must match the one you added in step 1
```

3. Clone the backend repository and run it
4. Use datagrip or any other tool to connect to the AWS database to verify that when you sign up your user information appears in the user table of the database.

# Notes
- There is one test that I commented out and I am aware of it. It is flaky so I need to fix it later on
- The **jacocoUnifiedTestReport** gradlew command seems to be broken as it does not reflect the proper coverage. If you run the unit tests, the user repository file will pretty good coverage but it falls drastically with the **jacocoUnifiedTestReport** command for some reason so I will investigate it further
